memcache/memcache_test.go

```diff
@@ -28,6 +28,8 @@ import (
 	"strings"
 	"testing"
 	"time"
+
+	"github.com/thanos-io/thanos/pkg/pool"
```

Comment:
Importing an actual pool implementation is more justifiable than an interface, but I still don't want to take the dependency on Thanos.

Can you use `sync.Pool` for the benchmark?